File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left_index = 0
        right_index = len(numbers) - 1

        for item in numbers:
            if numbers[left_index]  + numbers[right_index] == target:
                return [left_index + 1, right_index + 1]
            elif numbers[left_index] + numbers[right_index] > target:
                right_index -= 1
            else:
                left_index += 1
        return 0

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hashmap = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (hashmap.containsKey(complement)) {
                return new int[]{hashmap.get(complement), i};
            }
            hashmap.put(nums[i], i); // store value ➝ index
        }
        return new int[0];
    /**
    * Create a hashmap
    * Iterate through the array and add every element to the hashmap
    * Compute the target - current hashmap element, to find other num needed
    * Return the position/key of the two elements in the hashmap 
    */

    }
}
